### PR TITLE
std.concurrency housekeeping

### PR DESCRIPTION
**Pull request by burner**

### CODE REVIEW

File: std/concurrency.d [hunk 1]
```diff
@@ -693,18 +693,18 @@ body
 
 unittest
 {
-    assert( __traits( compiles,
+    static assert( __traits( compiles,
```
Comment (ZombineDev) [inline std/concurrency.d]: Maybe we should probably change `__traits(compiles, ...)` to a proper unittest. There was a call to action about that not long ago on the newsgroup. Anyway this change is an improvement.

File: std/concurrency.d [hunk 2]
```diff
@@ -1716,15 +1716,15 @@ void yield(T)(T value)
...
+/*version (Win64)
 {
     // fibers are broken on Win64
 }
 else version (Win32)
 {
     // fibers are broken in Win32 under server 2012: bug 13821
 }
-else unittest
+else */unittest
```
Comment (ZombineDev) [inline std/concurrency.d]: Hopefully we can remove this after the test completes on the Windows autotester.

Comment (burner) [inline std/concurrency.d]: that is my hope

File: std/concurrency.d (hunk 1 above)
Comment (burner) [inline std/concurrency.d]: what do you mean by proper unittest?

Comment (ZombineDev) [inline std/concurrency.d]: http://forum.dlang.org/post/n2027l$3r$1@digitalmars.com

Comment (burner) [inline std/concurrency.d]: thanks

**Comment by JackStouffer** [comment]
If the Windows tests pass, then LGTM

File: std/concurrency.d [hunk 3]
```diff
@@ -717,7 +717,7 @@ version (unittest)
 }
 unittest
 {
-    assert( __traits( compiles,
+    static assert( __traits( compiles,
```
Comment (JackStouffer) [inline std/concurrency.d]: This should also be changed eventually; probably best to do in a different PR.

Comment (burner) [inline std/concurrency.d]: I added that in this PR

**Comment by ZombineDev** [comment]
LGTM

**Comment by qznc** [comment]
LGTM

File: std/concurrency.d [hunk 4]
```diff
@@ -1716,15 +1716,7 @@ void yield(T)(T value)
...
-{
-    // fibers are broken on Win64
-}
-else version (Win32)
-{
-    // fibers are broken in Win32 under server 2012: bug 13821
-}
-else unittest
+unittest
 {
```
Comment (schveiguy) [inline std/concurrency.d]: I'm not sure that bug 13821 is fixed. If it is, the bug should be closed. I'm not sure if Brad moved the tester to 2012. Otherwise, this should be left the way it was.

Comment (wilzbach) [inline std/concurrency.d]: I'm not sure that bug 13821 is fixed. If it is, the bug should be closed. I'm not sure if Brad moved the tester to 2012. Otherwise, this should be left the way it was.

@burner did you find out details about this? Otherwise how about removing this change from the PR, so that we can move forward with the other housekeeping work? :)

Comment (burner) [inline std/concurrency.d]: https://issues.dlang.org/show_bug.cgi?id=13821

Comment (wilzbach) [inline std/concurrency.d]: > I'm not sure if Brad moved the tester to 2012

From the threads it's not clear whether @braddr managed to make the move:

https://github.com/dlang/druntime/pull/150 
http://forum.dlang.org/post/5480C893.4080507@puremagic.com

> I know this is ancient code, but it's failing on windows server 2012 for 32 bit windows apps.

Comment (burner) [inline std/concurrency.d]: the PR got merged and this PR passes the autotester. So I would conclude it is fixed.

**Comment by andralex** [comment]
thx!

**Comment by andralex** [comment]
Auto-merge toggled on

**Comment by burner** [comment]
thx
